src/lib/permissions.js

```diff
@@ -13,8 +13,10 @@ export const isRoleGreater = (role1, role2) =>
 export const hasRoleAtLeast = (hasRole, wantsRole) =>
   ROLE_HIERARCHY.indexOf(hasRole) >= ROLE_HIERARCHY.indexOf(wantsRole);
 
-export const getHighestRole = roles =>
-  roles.sort(isRoleGreater)[roles.length - 1];
+export const getHighestRole = roles => {
```

Comment:
make a copy so we don't copy something that came from `props`

Reply:
It looks like you added tempRoles.  Is this solely to remove an error message within development?  Or is there a larger function here that's being updated?

Reply:
`tempRoles` is  is a variable to hold a deep copy of roles.

1. Practical reason: with the updates, the code barfs when we try to modify a parameter of a react component. And by `barf` I mean it's a runtime error.
2. Theoretical reason: don't modify your function's parameters!